Simplenote/Note+Simplenote.swift

```diff
@@ -5,6 +5,21 @@ import Foundation
 //
 extension Note {
 
+    /// Returns the Creation / Modification date for a given SortMode
+    ///
+    func date(for sortMode: SortMode) -> Date? {
+        switch sortMode {
+        case .alphabeticallyAscending, .alphabeticallyDescending:
+            return nil
+
+        case .createdNewest, .createdOldest:
+            return creationDate
+
+        case .modifiedNewest, .modifiedOldest:
+            return modificationDate
+        }
+    }
```

Comment:
I don't think that the logic to select a date based on the sort mode belongs in the domain model. 
In my opinion it is something specific to how UI works and should be somewhere in the UI layer.

I have several ideas:
1. Keep it in the Note extension, make the extension private and move it to `NoteListViewController`
2. Move the code to `bodyPrefix` in `NoteListViewController`
3. Create a separate object that would format the date based on sortMode. For example, like this:

```
struct NoteListPrefixFormatter {
    func prefix(from note: Note, for sortMode: SortMode) -> String? {
        let date: Date? = {
            switch sortMode {
            case .alphabeticallyAscending, .alphabeticallyDescending:
                return nil

            case .createdNewest, .createdOldest:
                return note.creationDate

            case .modifiedNewest, .modifiedOldest:
                return note.modificationDate
            }
        }()

        if let date = date {
            return DateFormatter.notesFormatter.string(from: date)
        }

        return nil
    }
}

extension NoteListViewController {
    func bodyPrefix(for note: Note) -> String? {
         guard isSearching else {
             return nil
         }

        return NoteListPrefixFormatter().prefix(from: note, for: listController.sortMode)
     }
}
```
 
Because `NoteListPrefixFormatter` is a struct and doesn't have large instance variable, it can be created on stack and it's basically free to create and destroy it like this.